src/compare.c

```diff
@@ -0,0 +1,123 @@
+/*
+ * Compile with:
+ *
+ *      gcc -g -Wall compare.c `pkg-config vips --cflags --libs` -o compare
+ *
+ * Run with:
+ *
+ *      ./compare actual expected
+ *
+ * Examples:
+ *
+ *      ./compare actual.png expected.png
+ */
+
+#include <stdio.h>
+#include <vips/vips.h>
+
+
+/*
+  Compare images `actual` and `expected` and return mean square error.
+ */
+int Compare(VipsObject *context, VipsImage *actual, VipsImage *expected, double *out) {
+  if (actual->Bands != expected->Bands)
+    return -1;
+
+  if (actual->Type != expected->Type)
+    return -1;
+
+  VipsImage *difference;
+  VipsImage *stats;
+  if (vips_subtract(actual, expected, &difference, NULL) ||
+      vips_stats(difference, &stats, NULL))
+    return -1;
+
+  vips_object_local(context, difference);
+
+  printf("bands: %d\n", stats->Bands);
+  printf("bandformat: %d\n", stats->BandFmt);
+  printf("width: %d\n", stats->Xsize);
+  printf("height: %d\n", stats->Ysize);
+  printf("min: %f\n", (double) stats->data[0]);
+  printf("max: %f\n", (double) stats->data[1]);
+  printf("sum: %f\n", (double) stats->data[2]);
+  printf("sum2: %f\n", (double) stats->data[3]);
+  printf("avg: %f\n", (double) stats->data[4]);
+  printf("sd: %f\n", (double) stats->data[5]);
+  printf("xmin: %f\n", (double) stats->data[6]);
+  printf("ymin: %f\n", (double) stats->data[7]);
+  printf("xmax: %f\n", (double) stats->data[8]);
+  printf("ymax: %f\n", (double) stats->data[9]);
```

Comment:
What’s the right way to read the values from `vips_stats`? If this is correct, why I am only getting `0` for `sum2` when running it with two different images?